features/step_definitions/ab_testing_steps.rb

```diff
@@ -13,7 +13,11 @@
 end
 
 Given(/^I am in the "(A|B)" group for "(.*)" AB testing$/) do |test_group, ab_test|
+  # Set BOTH the cookie and the header, because:
+  # - On development/int/staging, cookies don't work so use the header directly
+  # - On production, the header is stripped, but cookies work
```

Comment:
Strictly speaking it's not stripped, it's just overwritten either by the existing cookie value or a random new value. (I know that's pedantic but the CDN config is confusing enough as it is, so I think it's worth making sure we describe it correctly.)